hedera-node/src/main/java/com/hedera/services/txns/schedule/ScheduleCreateTransitionLogic.java

```diff
@@ -50,12 +59,47 @@ public void doStateTransition() {
         }
     }
 
-    private void transitionFor(ScheduleCreateTransactionBody op) {
-        // TODO: Implement transitionFor() functionality
+    private void transitionFor(ScheduleCreateTransactionBody op) throws DecoderException {
+        var schedule = store.getScheduleIDByTransactionBody(op.getTransactionBody().toByteArray());
+        if (schedule.isEmpty()) {
```

Comment:
we have to slightly change the logic so that we do `createProvisionally` only when: `store.getScheduleIDByTransactionBody` returns empty or **if it returns tx but** `store.get(scheduleId).payer() != scheduled tx payer`

If it will be easier, we could change the `getScheduleIDByTransactionBody` method to `getScheduleID` and do:
```Java
var scheduledTXPayer = op.hasPayer() ? op.getPayer() : txnCtx.activePayer();
var scheduledTX = store.getScheduleID(txBody, scheduledTXPayer);

if (schedule.isEmpty() {
 // todo create
}
```